### PR TITLE
Added UT for Order controller.

### PR DESCRIPTION
**Pull request by yashgit891**
### - Link added should be a valid URL and accessible throughout the org.
   ### - If the branch name contains hotfix / revert by default the BVT workflow check will pass.

| Test Case Document URL                        |
|-----------------------------------------------|
| Please paste test case document link here.... |

### CODE REVIEW

File: Test/Unit/OrderControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+            Razorpay\Api\Order::class
+        );
+        $this->webhookApi = \Mockery::mock(
+            Razorpay\Api\Webhook::class
+        );
+        $this->requestApi = \Mockery::mock(
+            Razorpay\Api\Request::class
+        );
+        $this->apiError = \Mockery::mock(
+            Razorpay\Api\Errors\Error::class,['Test Api error message', 0, 0]
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php]: comma after space

File: Test/Unit/OrderControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+        $this->order->shouldReceive('getGrandTotal')->andReturn(1000);
+        $this->order->shouldReceive('getIncrementId')->andReturn('000012');
+        $this->order->shouldReceive('getOrderCurrencyCode')->andReturn('INR');
+        $this->order->shouldReceive('getEntityId')->andReturn('000012');
+        $this->order->shouldReceive('getPublicRazorpayApiInstance')->andReturn($this->api);
+
+        $this->checkoutSession->shouldReceive('getLastRealOrder')->andReturn($this->order);
+
+        $this->webhookData = ['entity' => 'collection',
+                            'count' => 1,
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php]: Align content accordingly

File: Test/Unit/OrderControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+                                    'active' => true,
+                                    'events' => [
+                                        'payment.authorized' => true,
+                                        'order.paid' => true,
+                                    ]
+                                ],
+                            ]
+                            ];
+        
+        $this->webhookData2 = ['entity' => 'collection',
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php]: Align content accordingly

File: Test/Unit/OrderControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+                                    'entity' => 'webhook',
+                                    'active' => true,
+                                    'events' => [
+                                        'payment.authorized' => true,
+                                        'order.paid' => true,
+                                    ]
+                                ],
+                            ]
+                            ];
+        $this->orderData = [
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php]: Align content accordingly

File: Test/Unit/OrderControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+        ];
+        $this->merchantPreferences = ["options" => [
+            "image"=> "https://cdn.razorpay.com/logos/IjFWzUIxXibcjw_medium.jpeg",
+            "redirect"=> true
+        ]];
+
+        $this->order->rzp = $this->api;
+        $this->resultFactory = $this->context->getResultFactory();
+        $this->order->setMockInit($this->_objectManager, $this->resultFactory);
+    }
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php:246]: Add space after function ends

File: Test/Unit/OrderControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+        $this->orderApi->shouldReceive('create')->andThrow($this->apiError);
+        $this->webhookApi->shouldReceive('edit')->andThrow($this->apiError);
+        $this->webhookApi->shouldReceive('create')->andThrow($this->apiError);
+        $this->webhookApi->shouldReceive('all')->with(['count' => 10, 'skip' => 0])->andReturn($this->webhookData2);
+
+        $this->requestApi->shouldReceive('request')->with("GET", "preferences")->andReturn($this->merchantPreferences);
+
+        $response = $this->order->execute();
+        $expectedResponse = '{"message":"Test Api error message","parameters":[]}';
+        $this->assertSame($expectedResponse, $this->getProperty($response, 'json'));
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php:304]: New line before all assert Statements

File: Test/Unit/OrderControllerTest.php [hunk 7]
```diff
@@ -0,0 +1,363 @@
+<?php
+declare(strict_types=1);
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php:3]: Add line before and after declare

File: Test/Unit/OrderControllerTest.php [hunk 8]
```diff
@@ -0,0 +1,363 @@
...
+use Razorpay\Magento\Test\MockFactory\MockApi;
+use Razorpay\Api\Api;
+use Magento\Framework\Controller\Result\Json;
+use Psr\Log\Test\TestLogger;
+
+/**
+ * @covers Razorpay\Magento\Controller\Payment\Order
+ */
+
+class OrderControllerTest extends TestCase {
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php]: Move flower bracket to next line and remove blank line before public

File: Test/Unit/OrderControllerTest.php [hunk 9]
```diff
@@ -0,0 +1,363 @@
...
+            Magento\Framework\Controller\ResultFactory::class
+        )->makePartial()->shouldAllowMockingProtectedMethods();
+
+        $this->serializerJson = new \Magento\Framework\Serialize\Serializer\Json;
+
+        $this->translateInline = \Mockery::mock(
+            \Magento\Framework\Translate\InlineInterface::class
+        );
+
+        $this->json = new Json($this->translateInline,
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php]: Move $this->serializerJson to same line

File: Test/Unit/OrderControllerTest.php [hunk 10]
```diff
@@ -0,0 +1,363 @@
...
+                case 'webhook':
+                    return $this->webhookApi;
+                case 'order':
+                    return $this->orderApi;
+                case 'request':
+                    return $this->requestApi;
+            }
+        });
+
+        $this->order = \Mockery::mock(Razorpay\Magento\Controller\Payment\Order::class, [$this->context,
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php]: Move array to next line

File: Test/Unit/OrderControllerTest.php [hunk 11]
```diff
@@ -0,0 +1,363 @@
...
+
+        $this->order->shouldReceive('getGrandTotal')->andReturn(1000);
+        $this->order->shouldReceive('getIncrementId')->andReturn('000012');
+        $this->order->shouldReceive('getOrderCurrencyCode')->andReturn('INR');
+        $this->order->shouldReceive('getEntityId')->andReturn('000012');
+        $this->order->shouldReceive('getPublicRazorpayApiInstance')->andReturn($this->api);
+
+        $this->checkoutSession->shouldReceive('getLastRealOrder')->andReturn($this->order);
+
+        $this->webhookData = ['entity' => 'collection',
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php]: Move array elements to next line

File: Test/Unit/OrderControllerTest.php [hunk 12]
```diff
@@ -0,0 +1,363 @@
...
+        
+        $this->setProperty($this->order, '_objectManager', $this->_objectManager);
+        $this->setProperty($this->order, 'resultFactory', $this->resultFactory);
+        
+        $_SERVER['HTTP_REFERER'] = 'Test Referrer URL';
+    }
+
+    function getProperty($object, $propertyName)
+    {
+        $reflection = new \ReflectionClass($object);
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php:250]: Add spaces

File: Test/Unit/OrderControllerTest.php [hunk 13]
```diff
@@ -0,0 +1,363 @@
...
+        $property = $reflection->getProperty($propertyName);
+        $property->setAccessible(true);
+        return $property->getValue($object);
+    }
+
+    function setProperty($object, $propertyName, $value)
+    {
+        $reflection = new \ReflectionClass($object);
```
Comment (garimaprachi) [inline Test/Unit/OrderControllerTest.php:260]: Remove redundant code

**Review by garimaprachi — COMMENTED** [review]
LGTM

**Review by abdulwahidsharief — APPROVED** [review]
LGTM

**Pull request merged**